src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1460 @@
+        self,
+        input_ids,
+        pixel_values,
+        attention_mask,
+        output_hidden_states=None,
+        output_attentions=None,
+        return_dict=None,
+        labels=None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3Processor, Beit3ForImagesAndTextClassification
+        >>> from PIL import Image
+        >>> import requests
+        >>> import torch
+
+        >>> model = Beit3ForImagesAndTextClassification.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+
+        >>> right_image_url = "https://i.ytimg.com/vi/DtTND8frecg/hqdefault.jpg"
+        >>> right_image = Image.open(requests.get(right_image_url, stream=True).raw)
+        >>> left_image_url = "https://cevgroup.org/wp-content/uploads/2018/10/WAP-7_class_electric_locomotive_of_Indian_Railways.jpg"
+        >>> left_image = Image.open(requests.get(left_image_url, stream=True).raw)
+
+        >>> text = "Power lines can be seen above the train in the image on the right."
+
+        >>> processor = Beit3Processor.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+        >>> inputs = processor(text=text, images=[left_image, right_image], return_tensors="pt")
+        >>> outputs = model(
+        ...     input_ids=inputs["input_ids"],
+        ...     pixel_values=inputs["pixel_values"].unsqueeze(0),
+        ...     attention_mask=torch.ones(inputs["input_ids"].shape),
+        ... )
+        >>> predicted_class_idx = outputs.logits.argmax(-1).item()
+        >>> predicted_class = model.config.id2label[predicted_class_idx]
+        >>> print("Predicted class:", predicted_class)
+        Predicted class: False
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        batch_size = input_ids.size()[0]
+        image1_values, image2_values = pixel_values.split(1, dim=1)
+        image1_values = image1_values.squeeze(1)
+        image2_values = image2_values.squeeze(1)
+        vision_input = torch.cat((image1_values, image2_values), dim=0)
+        language_input = torch.cat((input_ids, input_ids), dim=0)
+        attention_mask = torch.cat((attention_mask, attention_mask), dim=0)
+
+        outputs = self.beit3(
+            input_ids=language_input,
+            pixel_values=vision_input,
+            attention_mask=attention_mask,
+            output_hidden_states=output_hidden_states,
+            output_attentions=output_attentions,
+            return_dict=return_dict,
+        )
+        last_hidden_state = outputs[0]
+
+        if input_ids is None:
+            multiway_split_position = -1
+        elif pixel_values is None:
+            multiway_split_position = 0
+        else:
+            multiway_split_position = (self.config.image_size // self.config.patch_size) ** 2 + 1
+
+        vision_cls = last_hidden_state[:, 0, :]
+        language_cls = last_hidden_state[:, multiway_split_position, :]
```

Comment:
This means that the cls token representation taken for `language_cls` is the last token representation for the image inputs if `input_ids` is None. Is this what we want? 

Likewise if `pixel_values` is `None`, then we still have a `vision_cls` representation which is from the input ids.

Reply:
Good catch, input_ids and pixel_values cannot be none. Removing the if condition above.